File: iOS/Fs/EvolutionFs/ViewControllers/HomeViewController.fs
namespace EvolutionFs

open System
open System.Linq
open System.Drawing
open System.Collections.Generic

open MonoTouch.UIKit
open MonoTouch.Foundation
open GAColor

[<AllowNullLiteral>]
type HomeCell() as this =
    inherit UITableViewCell()

    do this.SelectionStyle <- UITableViewCellSelectionStyle.None
       this.Accessory <- UITableViewCellAccessory.DisclosureIndicator
       this.TextLabel.TextColor <- GAColor.EvoGreen.ToUIColor()
    
    member this.Text 
        with get() = this.TextLabel.Text
        and set(value) = this.TextLabel.Text <- value
    
    static member val Key = "c" with get

type HomeListViewSource(optionSelected:string->unit) =
    inherit UITableViewSource()

    member val Items = [||] with get,set

    override this.RowsInSection (tableview, section) =
        if this.Items = [||] then 1 else this.Items.Length

    override this.RowSelected (tableView, indexPath) =
        if not (this.Items = [||]) then
            optionSelected (this.Items.[indexPath.Row])

    override this.GetCell (tableView, indexPath) =
        let cell = match tableView.DequeueReusableCell(HomeCell.Key) with
            | :? HomeCell as cell when not (cell = null) -> cell
            | _ -> new HomeCell()

        cell.Text <- this.Items.[indexPath.Row]
        cell :> UITableViewCell


type HomeViewController() as this =
    inherit UITableViewController()

    let imageView = new JBKenBurnsView()
    let cellHeight = 55.f
    let source = new HomeListViewSource(fun x-> this.OptionSelected(x))

    let LoadTable = 
        this.TableView.RowHeight <- cellHeight
        this.TableView.Source <- source :> UITableViewSource
        source.Items <- [|"Evolution of Color"|]
        this.TableView.ReloadData()
    
    do
        this.Title <- "Evolution and Artificial Life"
        this.NavigationItem.BackBarButtonItem <- new UIBarButtonItem ("", UIBarButtonItemStyle.Plain, handler=null)
        this.View.Frame <- UIScreen.MainScreen.Bounds
        this.View.BackgroundColor <- UIColor.White
        this.View.AutoresizingMask <- UIViewAutoresizing.FlexibleWidth

        imageView.Frame <- new RectangleF(0.f, -20.f, 320.f, this.View.Bounds.Height - (2.f * cellHeight) - 75.f)
        imageView.Images <- new List<UIImage>()
        imageView.Images.Add(UIImage.FromFile("Assets/splash_square.png"))
        imageView.Images.Add(UIImage.FromFile("Assets/splash_square.png"))
        imageView.UserInteractionEnabled <- false

        this.TableView.TableHeaderView <- new UIView(new RectangleF(0.f, 0.f, this.View.Bounds.Width, imageView.Frame.Bottom))
        this.View.AddSubview(imageView)
        LoadTable 

    member val ImageWidth = UIScreen.MainScreen.Bounds.Width * UIScreen.MainScreen.Scale with get
    member val OptionSelected = (fun (x:string)->()) with get,set

    override this.ViewWillAppear animated =
        base.ViewWillAppear (animated)
        imageView.Animate()

File: iOS/Cs/Evolution.iOS/ViewControllers/HomeViewController.cs
/*Copyright 2014 All Good People LLC

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.*/

using Evolution.Shared;
using MonoTouch.Foundation;
using MonoTouch.UIKit;
using System;
using System.Collections.Generic;
using System.Drawing;

namespace Evolution.iOS
{
    public class HomeViewController : UITableViewController
    {
        public event Action GoToSettingsVC = delegate { };

        JBKenBurnsView imageView;
        HomeTableSource source;
        RectangleF bounds;
        int CellHeight = 55;

        public HomeViewController(Action GoToSettingsVC)
        {
            Title = "Evolution++";
            NavigationItem.BackBarButtonItem = new UIBarButtonItem("", UIBarButtonItemStyle.Plain, handler: null);

            this.GoToSettingsVC = GoToSettingsVC;

            LoadTable();
        }

        private void LoadTable()
        {
            TableView.RowHeight = CellHeight;
            TableView.Source = source = new HomeTableSource();
            source.GoToActions = new Action[1] { this.GoToSettingsVC };
            source.Items = new List<string>() { "Evolution of Color" };
            TableView.ReloadData();
        }

        public override void ViewDidLoad()
        {
            base.ViewDidLoad();

            View.Frame = this.bounds = UIScreen.MainScreen.Bounds;
            View.BackgroundColor = UIColor.White;
            View.AutoresizingMask = UIViewAutoresizing.FlexibleWidth | UIViewAutoresizing.FlexibleHeight;

            imageView = new JBKenBurnsView
            {
                Frame = new RectangleF(0, -20, 320, bounds.Height - (2 * CellHeight) - 75),
                Images = new List<UIImage> {
					UIImage.FromFile("Assets/splash_square.png"),
					UIImage.FromFile("Assets/splash_square.png"),
				},
                UserInteractionEnabled = false,
            };
            TableView.TableHeaderView = new UIView(new RectangleF(0, 0, bounds.Width, imageView.Frame.Bottom));
            View.AddSubview(imageView);
        }

        public override void ViewWillAppear(bool animated)
        {
            base.ViewWillAppear(animated);
			if (AppDelegate.Shared.adBanner != null)
			{
				//AppDelegate.Shared.adBanner.RemoveFromSuperview ();
				AppDelegate.Shared.adBanner.Hidden = true;
			}
            imageView.Animate();
        }
    }
}